segmentation/evaluate.py

```diff
@@ -0,0 +1,88 @@
+import argparse
+import base64
+import io
+from pathlib import Path
+from typing import List, Iterator
+import time
+
+import numpy as np
+import pandas as pd
+from PIL import Image
+import requests
+
+from telesto.apps.segmentation import image_to_base64
+
+
+def jaccard(y_pred: np.ndarray, y_true: np.ndarray) -> float:
+    intersection = (y_pred * y_true).sum()
+    union_plus_intersection = y_true.sum() + y_pred.sum()
+    return intersection / (union_plus_intersection - intersection)
+
+
+def estimate_metric(
+    pred_list: Iterator[np.ndarray], label_list: Iterator[np.ndarray], class_n: int
+) -> float:
+    metrics_per_image = []
+    for preds, labels in zip(pred_list, label_list):
+        metrics_per_class = {cl: jaccard(preds == cl, labels == cl) for cl in range(class_n)}
+        metric_avg = sum(metrics_per_class.values()) / len(metrics_per_class)
+        metrics_per_image.append(metric_avg)
+    return sum(metrics_per_image) / len(metrics_per_image)
```

Comment:
There are some issues here.

What is the input of `jaccard`? There are two possible formats for predictions and ground truths.
1. `np.ndarray` of shape `(height, width)`, each element containing the integer class label.
2. `np.ndarray` of shape `(num_classes, height, width)`, each pixel is a probability map. (For the ground truth, it is a binary image.)

The `jaccard` function is correct when using the no. 2 format, while the `estimate_metric` seems to work with format no. 1. I think the `estimate_metric` function is not needed.